OrbitCore/SamplingProfiler.cpp

```diff
@@ -426,7 +426,9 @@ void SamplingProfiler::UpdateAddressInfo(uint64_t address) {
   if (function != nullptr) {
     function_address = function->GetVirtualAddress();
     function_name = function->PrettyName();
-    address_info->function_name = function_name;
+    if (address_info != nullptr) {
```

Comment:
It doesn't change anything but for readability consider moving the update down to where the other updates happen (below line 444). Then you have a clear structure where here you are just filling function_address and function_name with the correct values and then you do all necessary updates to the data structures below. Will IMO be easier to maintain later.

Reply:
done.